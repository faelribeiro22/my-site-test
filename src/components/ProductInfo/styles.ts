'use client'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    background: ${theme.colors.white};
    padding: ${theme.spacings.small};
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray};
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      max-width: 77rem;
    `}
  `}
`

export const ButtonsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;

    > button {
      width: 100%;
      margin-bottom: ${theme.spacings.xxsmall};
    }

    ${media.greaterThan('medium')`
      flex-direction: row-reverse;

      > button {
        width: initial;
        margin-bottom: 0;
      }
    `}
  `}
`

export const ContainerImage = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 23rem;
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      height: 58rem;
    `}
  `}
`
