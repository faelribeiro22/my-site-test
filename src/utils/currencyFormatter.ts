export function currencyFormat(valor: number) {
  return valor.toLocaleString("pt-BR", { style: "currency", currency: "BRL" });
}
