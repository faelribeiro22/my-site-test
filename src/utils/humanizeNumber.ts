export function humanizeNumber(value: number) {
  if (value >= 1_000_000) {
    return (value / 1_000_000).toFixed(1).replace(".0", "") + "M";
  }
  if (value >= 1_000) {
    return (value / 1_000).toFixed(1).replace(".0", "") + "k";
  }
  return value.toString();
}
