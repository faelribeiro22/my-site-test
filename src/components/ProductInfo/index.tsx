import Button from '../Button'
import Heading from '../Heading'
import { AddShoppingCart } from '@styled-icons/material-outlined'
import * as S from './styles'
import Image from 'next/image'
import { BuyBox, Price } from '../ProductCard/styles'

export type ProductInfoProps = {
  title: string
  description: string
  price: string
  image: string
  category: string
}

const ProductInfo = ({ title, description, price, image }: ProductInfoProps) => (
  <S.Wrapper>
    <Heading color="black" $lineBottom>
      {title}
    </Heading>
    <S.ContainerImage>
      <Image src={image} alt={title} fill={true}/>
    </S.ContainerImage>
    <S.Description>{description}</S.Description>

    <BuyBox>
      <Price>{`${price}`}</Price>
      <Button icon={<AddShoppingCart />} size="small">
        Add to cart
      </Button>
    </BuyBox>
  </S.Wrapper>
)

export default ProductInfo
