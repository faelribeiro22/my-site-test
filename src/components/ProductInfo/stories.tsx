import { Meta, StoryObj } from '@storybook/react'
import ProductInfo, { ProductInfoProps } from '.'

const meta: Meta<ProductInfoProps> = {
  title: 'Product/ProductInfo',
  component: ProductInfo,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  },
  argTypes: {
    title: {
      type: 'string'
    },
    description: {
      type: 'string'
    },
    price: {
      type: 'string'
    },
    image: {
      type: 'string'
    }
  }
}

export default meta

export const Default: StoryObj = {}

export const Basic: StoryObj<ProductInfoProps> = {
  render: (args) => (
    <div style={{ maxWidth: '144rem', padding: '1.5rem' }}>
      <ProductInfo {...args} />
    </div>
  )
}

Basic.args = {
  title: 'A new Phone',
  description: 'A new phone with the best features',
  image: 'https://images.igdb.com/igdb/image/upload/t_cover_big/co1jyc.jpg',
  price: 'R$ 235,00'
}
