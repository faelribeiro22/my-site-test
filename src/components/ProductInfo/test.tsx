import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import ProductInfo from '.'
import { currencyFormat } from '../../utils/currencyFormatter'

const props = {
  id: '1',
  title: 'product Title 2',
  description: 'the new product new',
  price: currencyFormat(210),
  image: 'https://source.unsplash.com/user/willianjusten/300x140',
  category: 'category'
}

describe('<ProductInfo />', () => {
  it('should render product information', () => {
    const { container } = renderWithTheme(<ProductInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /product title 2/i })
    ).toBeInTheDocument()
    expect(screen.getByText('R$ 210,00')).toBeInTheDocument()
    expect(screen.getByText(/the new product new/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render buttons', () => {
    renderWithTheme(<ProductInfo {...props} />)
    expect(
      screen.getByRole('button', { name: /add to cart/i })
    ).toBeInTheDocument()
  })
})
