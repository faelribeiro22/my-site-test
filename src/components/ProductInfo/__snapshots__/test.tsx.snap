// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo /> should render product information 1`] = `
.c6 {
  display: inline-flex;
  justify-content: center;
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  height: 3rem;
  font-size: 1.2rem;
}

.c6:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c1::after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -1rem;
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  position: relative;
  background: #FAFAFA;
  padding: 2.4rem;
}

.c3 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
}

.c2 {
  position: relative;
  width: 100%;
  height: 23rem;
  margin-bottom: 2.4rem;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

.c5 {
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  align-items: center;
  color: #FAFAFA;
  padding: 0 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  margin-right: calc(0.8rem / 2);
}

@media (min-width: 768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width: 768px) {
  .c3 {
    max-width: 77rem;
  }
}

@media (min-width: 768px) {
  .c2 {
    height: 58rem;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
  >
    product Title 2
  </h2>
  <div
    class="c2"
  >
    <img
      alt="product Title 2"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F300x140&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <p
    class="c3"
  >
    the new product new
  </p>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      R$Â 210,00
    </div>
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
  </div>
</div>
`;
